packages/video_player/video_player_web/lib/video_player_web.dart

```diff
@@ -42,8 +42,27 @@ class VideoPlayerPlugin extends VideoPlayerPlatform {
     final int textureId = _textureCounter;
     _textureCounter++;
 
+    Uri uri;
+    switch (dataSource.sourceType) {
+      case DataSourceType.network:
+        uri = Uri.parse(dataSource.uri);
+        break;
+      case DataSourceType.asset:
+        // 'webOnlyAssetManager' is only in the web version of dart:ui
+        // ignore: undefined_prefixed_name
+        String assetUrl = ui.webOnlyAssetManager.getAssetUrl(dataSource.asset);
+        if (dataSource.package != null && dataSource.package.isNotEmpty) {
+          assetUrl = 'packages/${dataSource.package}/$assetUrl';
+        }
+        uri = Uri.parse(assetUrl);
+        break;
+      default:
```

Comment:
nit: I think it would be a bit clearer and more future-proof to put a `DataSourceType.file` case explicitly here.

Reply:
Done.